lib/inspec/rspec_json_formatter.rb

```diff
@@ -278,272 +268,437 @@ class InspecRspecCli < InspecRspecJson # rubocop:disable Metrics/ClassLength
-          (skips.length > 0) ? "#{skips.length} skipped" : nil,
-        ].compact.join(' ')
+        # determine title
+        test_result = test[:skip_message] || test[:code_desc].split[2..-1].join(' ')
+        # show error message
+        test_result += "\n" + test[:message] unless test[:message].nil?
       end
-    if suffix == ''
-      title
-    else
-      title + ' (' + suffix + ')'
+      status_indicator = test[:status_type]
+      print_line(
+        color:      COLORS[status_indicator] || '',
+        indicator:  INDICATORS['small'] + INDICATORS[status_indicator] || INDICATORS['unknown'],
+        summary:    format_lines(test_result, INDICATORS['empty']),
+        id:         control_id,
+        profile:    control[:profile_id],
+      )
     end
   end
 
+  def print_profile_summary
+    summary = profile_summary
+
+    s = format('Profile Summary: %s%d successful%s, %s%d failures%s, %s%d skipped%s',
+               COLORS['passed'], summary['passed'], COLORS['reset'],
+               COLORS['failed'], summary['failed']['total'], COLORS['reset'],
+               COLORS['skipped'], summary['skipped'], COLORS['reset'])
+    output.puts(s) if summary['total'] > 0
+  end
+
+  def print_tests_summary
+    summary = tests_summary
+
+    s = format('Test Summary: %s%d successful%s, %s%d failures%s, %s%d skipped%s',
+               COLORS['passed'], summary['passed'], COLORS['reset'],
+               COLORS['failed'], summary['failed'], COLORS['reset'],
+               COLORS['skipped'], summary['skipped'], COLORS['reset'])
+    output.puts(s)
+  end
+
+  # Formats the line (called from print_line)
   def format_line(fields)
-    @format.gsub(/%\w+/) do |x|
+    format = '%color%indicator%id%summary'
+    format.gsub(/%\w+/) do |x|
       term = x[1..-1]
       fields.key?(term.to_sym) ? fields[term.to_sym].to_s : x
-    end + @colors['reset']
+    end + COLORS['reset']
   end
 
+  # Prints line; used to print results
   def print_line(fields)
     output.puts(format_line(fields))
   end
 
+  # Helps formatting summary lines (called from within print_line arguments)
   def format_lines(lines, indentation)
     lines.gsub(/\n/, "\n" + indentation)
   end
 
-  def print_results(all)
-    all.each do |x|
-      test_status = x[:status_type]
-      test_color = @colors[test_status]
-      indicator = @indicators[x[:status]]
-      indicator = @indicators['empty'] if indicator.nil?
-      if x[:message]
-        msg = x[:code_desc] + "\n" + x[:message]
-      else
-        msg = x[:skip_message] || x[:code_desc]
-      end
-      print_line(
-        color:      test_color,
-        indicator:  @indicators['small'] + indicator,
-        summary:    format_lines(msg, @indicators['empty']),
-        id: nil, profile: nil
-      )
+  #
+  # This class wraps a control hash object to provide a useful inteface for
+  # maintaining the associated profile, ids, results, title, etc.
+  #
+  class Control
```

Comment:
Don't love that I used the name `Control` here but it is one within this context. I wouldn't want to confuse anyone here.